### PR TITLE
Update dependencies 2022-11

### PR DESCRIPTION
**Pull request by FinThunderstorm**
Updated dependencies. 

React not updated due to deprecation of `express-react-views`.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,15 +20,15 @@ services:
...
+      INITIAL_BUCKETS: 'local/exam-archive-local'
     volumes:
       - exam-archive-minio-data:/data
     ports:
       - 127.0.0.1:9001:9000
       - 127.0.0.1:9002:9001
 
   db:
-    image: postgres:11.16
+    image: postgres:15.0
```
Comment (r-jheikkila) [inline docker-compose.yml]: RDS instance is still 11 so might want to stay on the same version until we find the time to upgrade. Probably 11.17 or whatever is latest.

File: seeds/courses.js [hunk 2]
```diff
@@ -790,4 +790,9 @@ exports.seed = function (knex) {
         }
       ])
     })
+    .then(() => {
+      return knex.raw(
```
Comment (r-jheikkila) [inline seeds/courses.js:794]: Could separate into own commit as this seems like its own bugfix (fix id sequence because inserts set explicit ids which does not increment sequence) 😁

File: seeds/exams.js [hunk 3]
```diff
@@ -7406,4 +7406,9 @@ exports.seed = function (knex) {
         }
       ])
     })
+    .then(() => {
+      return knex.raw(
+        "SELECT setval('exams_id_seq', (SELECT MAX(id) FROM exams), true);"
```
Comment (r-jheikkila) [inline seeds/exams.js:7411]: Same

File: scripts/seed-db-from-files.js [hunk 4]
```diff
@@ -106,7 +106,10 @@ const start = async (sourceDirectory, mod, markDirty) => {
   console.group(`Inserting ${courses.length} courses to db...`)
   /** @type {{id: number, name: string}[]} */
   const insertedCourses = await knex
-    .batchInsert('courses', courses.map(({ name }) => ({ name })))
```
Comment (r-jheikkila) [inline scripts/seed-db-from-files.js:109]: Would it be too much work to separate ’run prettier’ into own commit so we can https://michaelheap.com/git-ignore-rev/ ?

File: src/api.ts [hunk 5]
```diff
@@ -50,7 +49,7 @@ api.get(
 
 api.post(
   '/courses/:courseId/rename',
-  bodyParser.json(),
+  express.json(),
```
Comment (r-jheikkila) [inline src/api.ts:52]: Good catch!

File: src/download.ts [hunk 6]
```diff
@@ -29,26 +18,22 @@ const getUrlExpirationTimestamp = () =>
...
-const createSignedS3Url = (exam: DbExam) =>
-  s3.getSignedUrlPromise('getObject', {
+const createSignedS3Url = (exam: DbExam) => {
+  const command = new GetObjectCommand({
     Bucket: config.AWS_S3_BUCKET_ID,
-    Key: exam.file_path,
-    Expires: getUrlExpirationTimestamp()
+    Key: exam.file_path
   })
+  return getSignedS3Url(s3, command, { expiresIn: oneDayToSeconds })
```
Comment (r-jheikkila) [inline src/download.ts:35]: Very nice lapiointi

File: src/index.ts [hunk 7]
```diff
@@ -175,6 +177,8 @@ app.use(async (req, res, next) => {
         : noRights
     } as AuthData
   } catch (e) {
+    if (!axios.isAxiosError(e))
```
Comment (r-jheikkila) [inline src/index.ts:180]: Very good catch. Would add console.error() and separate to own commit.

File: static/vendor/instantpage-5.1.0.js [hunk 8]
```diff
@@ -1,2 +1,151 @@
 /*! instant.page v5.1.0 - (C) 2019-2020 Alexandre Dieulot - https://instant.page/license */
```
Comment (r-jheikkila) [inline static/vendor/instantpage-5.1.0.js]: Accidental addition belonging to separate PR perhaps?

**Review by r-jheikkila — CHANGES_REQUESTED** [review]
Awesome stuff, I really appreciate the work! Left a few small comments. Let me know if I can help split stuff into commits, I think it’s really nice to have a minttu blame history.

File: src/index.ts (hunk 7 above)
Comment (FinThunderstorm) [reply src/index.ts:180]: This is needed for Typescript to compile and adding clean console.error() makes compiler to complain => did a little trick and added console.error().

File: static/vendor/instantpage-5.1.0.js (hunk 8 above)
Comment (FinThunderstorm) [reply static/vendor/instantpage-5.1.0.js]: prettier... :D

File: docker-compose.yml (hunk 1 above)
Comment (FinThunderstorm) [reply docker-compose.yml]: changed to 11.7

File: seeds/courses.js (hunk 2 above)
Comment (FinThunderstorm) [reply seeds/courses.js:794]: splitted

File: seeds/exams.js (hunk 3 above)
Comment (FinThunderstorm) [reply seeds/exams.js:7411]: splitted

File: scripts/seed-db-from-files.js (hunk 4 above)
Comment (FinThunderstorm) [reply scripts/seed-db-from-files.js:109]: prettier only stuff splitted into own commit

File: src/index.ts [hunk 9]
```diff
@@ -175,6 +177,14 @@ app.use(async (req, res, next) => {
...
+      return () => {
+        console.error(
+          'user-service getMe failed with error is not AxiosError',
+          e
+        )
+        res.status(500).render('error-user-service')
+      }
```
Comment (cxcorp) [inline src/index.ts:187]: I think this is just a no-op since it returns a function but handler return values aren't utilized in express, right?

**Comment by cxcorp** [comment]
Thanks so much for your contribution!
